src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,235 @@
-from typing import Tuple, Union
+from __future__ import annotations
+
+import warnings
+from typing import Optional, Union
+
+import numpy as np
 
 __all__ = ["TimeSteppingControl"]
 
 
 class TimeSteppingControl:
-    """Parent class for iteration-based time stepping control routine."""
+    """Parent class for iteration-based time-stepping control routine.
+
+    Parameters:
+        schedule: List containing the target times for the simulation.
+            Unless a constant time step is prescribed, the time-stepping algorithm will adapt
+            the time step so that the scheduled times are guaranteed to be hit.
+
+            The `schedule` list must contain minimally two items, corresponding to the
+            initial and final simulation times. Lists of length > 2 must contain strictly
+            increasing times. Examples of VALID inputs are: [0, 1], [0, 10, 30, 50], and
+            [0, 1*pp.HOUR, 3*pp.HOUR]. Examples of INVALID inputs are: [1], [1, 0], and
+            [0, 1, 1, 2].
```

Comment:
Not crucial: Yet, why not internally providing simple mechanisms which unify and sort the items? This would relax the requirements. Typical use case would be if one wants to add another prescribed time to a long list of times. Then searching for the right position is doable, but maybe not necessary.

Reply:
Interesting feature. This should doable, yes. Do we want something like this @keileg?

Reply:
We decided to keep it as it is, meaning , to require strictly increasing times.